### PR TITLE
Support all HTML tags with all React domProps

### PR DESCRIPTION
**Pull request by davesnx**
> #### Old issue: 
> First I thought the syntax would work like that:
> - `[%styled.div ""];` But there are a few limitations: Can't define dynamically the HTML tag. Add noise to the api, and It's a little bit complex to create those ppxs. Allows that  It would remove the ability to `styled(Component)`.
> - In styled-components they use "as" prop in the render. I like that and is reasonable simple to implement in the ppx.

Hi, 

This PR started as "I want to support all HTML elements", so I added a list and changed the ppx to be `[%styled.div]` and all the variants.

Later I decided to push a little forward to see how hard would be to add the domProps that React uses in their JSX to transform to dom attributes or event handlers internally.

Finally I end up with a rewrite of the module to bypass [@react.component] and just create the makeProps record.

I made it to this PR since it allows to think about the next steps to turn styled into a function and be ready to receive those makeProps. If it's a function we would be ready to get the ast and detect which props are passed to the StyledComponent and create the styled with those props (and add them to makeProps). (Exactly this issue https://github.com/davesnx/styled-ppx/issues/12)

So, this PR it would solve https://github.com/davesnx/styled-ppx/issues/44 and https://github.com/davesnx/styled-ppx/issues/26

In order to merge this PR we need to:
- [x] Create a binding to `React.createElement` https://es.reactjs.org/docs/react-api.html#createelement
- [x] Add a type to the module to hide the `setClassName` and the `React.createElement` binding. https://2ality.com/2017/12/modules-reasonml.html
- [x] Add mutable into className
- [x] Update README.md and ROADMAP.md
- [ ] Fix `styled.tag` ppx detection. Probably worth it adding another solution? https://github.com/paurkedal/ppx_regexp https://github.com/Drup/tyre
- [ ] Try to improve test native.
- [x] Create some snapshot tests in BuckleScript.

### CODE REVIEW

File: src/styled_ppx.re [hunk 1]
```diff
@@ -137,12 +259,12 @@ let moduleMapper = (_, _) => {
     switch (expr) {
     | {
         pmod_desc:
-          /* that are defined with the ppx [%styled] */
+          /* that are defined with a ppx like [%txt] */
           Pmod_extension((
-            {txt: "styled", _},
+            {txt, _},
```
Comment (davesnx) [inline src/styled_ppx.re]: Hey @jchavarri, do you know if this is performant? I'm mapping thought all the ppx that have a text and later on, doing the split.

So I'm not sure if it's even a good idea. WDYT?

Comment (davesnx) [reply src/styled_ppx.re]: I think it blocks other ppx to run actually 😢

**Comment by davesnx** [comment]
Probably worth it adding https://github.com/Drup/tyre to catch up `styled.tag` better.

File: src/styled_ppx.re (hunk 1 above)
Comment (jchavarri) [reply src/styled_ppx.re]: It shouldn't block other ppxs. I think the issue might be that the current code is "swallowing" the attributes of other ppxs.

If the ast returned respects other existing attributes and returns them, other ppxs should work ok 🤔

**Pull request merged**